packages/agreement-process/test/agreementService.integration.test.ts

```diff
@@ -1181,4 +1182,24 @@ describe("Agreement service", () => {
       );
     });
   });
+  describe("get agreement", () => {
+    it("should get an agreement", async () => {
+      const agreement: Agreement = buildAgreement(
+        generateId<EServiceId>(),
+        generateId<TenantId>(),
+        agreementState.draft
+      );
+      await addOneAgreement(agreement, postgresDB, agreements);
+
+      const result = await agreementService.getAgreementById(agreement.id);
+      expect(result).toEqual(agreement);
+    });
+
+    it("should throw an agreementNotFound error when the agreement does not exist", async () => {
+      const agreementId = generateId<AgreementId>();
+      await expect(
+        agreementService.getAgreementById(agreementId)
+      ).rejects.toThrowError(agreementNotFound(agreementId));
+    });
```

Comment:
For the error path: could we have a non-matching agreement in the readmodel?

Reply:
Makes sense, done!